hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterHeartbeatManager.java

```diff
@@ -160,7 +160,13 @@ public void handleHeartbeat(MembersViewMetadata senderMembersViewMetadata, Strin
 
         try {
             if (!clusterService.isJoined()) {
-                logger.fine("Ignoring heartbeat of sender: " + senderMembersViewMetadata + ", because node is not joined!");
+                if (clusterService.getThisUuid().equals(receiverUuid)) {
+                    logger.fine("Ignoring heartbeat of sender: " + senderMembersViewMetadata + ", because node is not joined!");
```

Comment:
I don't know the frequency of this code being called; but if it is frequent, it is best to add a logging guard to prevent litter.